packages/hosted-credit-card-integration/src/utils/useHostedCreditCard.tsx

```diff
@@ -0,0 +1,220 @@
+import {
+    CardInstrument,
+    CheckoutSelectors,
+    HostedFormOptions,
+    LanguageService,
+    PaymentMethod,
+} from '@bigcommerce/checkout-sdk';
+import { compact, forIn } from 'lodash';
+import React, { ReactNode, useCallback, useState } from 'react';
+
+import {
+    CreditCardInputStylesType,
+    getCreditCardInputStyles,
+    isInstrumentCardCodeRequiredSelector,
+    isInstrumentCardNumberRequiredSelector,
+} from '@bigcommerce/checkout/instrument-utils';
+import { PaymentFormService } from '@bigcommerce/checkout/payment-integration-api';
+
+import { HostedCreditCardValidation } from '../components';
+
+export interface UseHostedFormOptions {
+    checkoutState: CheckoutSelectors;
+    language: LanguageService;
+    method: PaymentMethod;
+    paymentForm: PaymentFormService;
+}
+
+export const useHostedCreditCard = ({
+    checkoutState,
+    method,
+    language,
+    paymentForm,
+}: UseHostedFormOptions) => {
+    const { setFieldTouched, setFieldValue, setSubmitted, submitForm } = paymentForm;
+    const {
+        config: { cardCode },
+    } = method;
+    const isCardCodeRequired = cardCode || cardCode === null;
```

Comment:
Is there any other way to tell if cardCode is quired?
Depending on its own value seems not logical.

Reply:
As far as I know we only can depend on this variable from config, the same logic we have in `HostedCreditCardPaymentMethod.tsx`